packages/SwingSet/src/kernel/kernel.js

```diff
@@ -1064,46 +1081,60 @@ export default function buildKernel(
   }
 
   let processQueueRunning;
-  async function tryProcessDeliveryMessage(message) {
+  async function tryProcessMessage(processMessage, message) {
     if (processQueueRunning) {
       console.error(`We're currently already running at`, processQueueRunning);
       assert.fail(X`Kernel reentrancy is forbidden`);
     }
     processQueueRunning = Error('here');
-    return processDeliveryMessage(message).finally(() => {
+    return processMessage(message).finally(() => {
       processQueueRunning = undefined;
     });
   }
 
   async function processAcceptanceMessage(message) {
     kdebug(`processAcceptanceQ ${JSON.stringify(message)}`);
     kdebug(legibilizeMessage(message));
-    if (processQueueRunning) {
```

Comment:
I'm guessing you determined that this extra check is unnecessary, and/or breaks with the new arrangement?

Reply:
No I refactored the re-entrancy check to share it with `processDeliveryMessage` in https://github.com/Agoric/agoric-sdk/pull/5252/commits/f57b3400a771062b2d83798f232b770c2c36fb5e